internal/dev_server/sdk/sdk_events.go

```diff
@@ -0,0 +1,105 @@
+package sdk
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/launchdarkly/ldcli/internal/dev_server/events"
+	"github.com/launchdarkly/ldcli/internal/dev_server/model"
+	"github.com/pkg/errors"
+)
+
+func newSdkEventObserver(updateChan chan<- Message, ctx context.Context) sdkEventObserver {
+	debugSessionKey := uuid.New().String()
+	db := model.EventStoreFromContext(ctx)
+	err := db.CreateDebugSession(ctx, debugSessionKey)
+	if err != nil {
+		log.Printf("sdkEventObserver: error writting debug session: %v", err)
+	}
+	return sdkEventObserver{
+		debugSessionKey: debugSessionKey,
+		ctx:             ctx,
+		updateChan:      updateChan,
+	}
+}
+
+type sdkEventObserver struct {
+	ctx             context.Context
+	debugSessionKey string
+	updateChan      chan<- Message
+}
+
+func (o sdkEventObserver) Handle(message interface{}) {
+	str, ok := message.(json.RawMessage)
+	if !ok {
+		return
+	}
+
+	event := events.Base{}
+	err := json.Unmarshal(str, &event)
+	if err != nil {
+		log.Printf("sdkEventObserver: error unmarshaling event: %v", err)
+		return
+	}
+
+	db := model.EventStoreFromContext(o.ctx)
+
+	err = db.WriteEvent(o.ctx, o.debugSessionKey, event.Kind, str)
+	if err != nil {
+		log.Printf("sdkEventObserver: error writting event: %v", err)
+		return
+	}
+
+	o.updateChan <- Message{Event: TYPE_PUT, Data: str}
+}
+
+var observers *model.Observers = model.NewObservers()
+
+func SdkEventsReceiveHandler(writer http.ResponseWriter, request *http.Request) {
+	bodyStr, err := io.ReadAll(request.Body)
+	if err != nil {
+		log.Printf("SdkEventsReceiveHandler: error reading request body: %v", err)
+		return
+	}
+
+	var arr []json.RawMessage
+	err = json.Unmarshal(bodyStr, &arr)
+
+	if err != nil {
+		log.Printf("SdkEventsReceiveHandler: error unmarshaling request body: %v", err)
+	}
+
+	for _, msg := range arr {
+		observers.Notify(msg)
+	}
+
+	writer.Header().Set("Content-Type", "application/json")
+	writer.WriteHeader(http.StatusAccepted)
+}
+
+func SdkEventsTeeHandler(writer http.ResponseWriter, request *http.Request) {
```

Comment:
ohhhh. I get what this is for now. I think we should move this to not be in the SDK package. Similarly for this observer.